static/js/protein/data_processing_utils.ts

```diff
@@ -0,0 +1,332 @@
+             if (n.property === "geneSymbol") {
+               for(const n1 of n.nodes) {
+                 for(const n2 of n1.neighbors) {
+                   if(n2.property === "referenceSNPClusterID") {
+                    if(n2.nodes[0].value !== null)
+                      {variant = n2.nodes[0].value;}
+                   } 
+                   else if(n2.property === "log2AllelicFoldChange") {
+                      if(n2.nodes[0].value !== null)
+                        {score = n2.nodes[0].value;}
+                   } 
+                   else if(n2.property === "log2AllelicFoldChangeConfidenceInterval") {
+                      if(n2.nodes[0].value !== null)
+                        {interval = n2.nodes[0].value;}
+                   } 
+                   else if(n2.property === "associatedTissue") {
+                    if(n2.nodes[0].value)
+                      {tissue = n2.nodes[0].value;}
+                   }
+                 }
+                  if (Array.isArray(objArray)) {
+                      objArray.push({id: variant, name: tissue, value: score, interval: interval});
+                  } else {
+                    console.log('arr variable does not store an array');
+                  }
+                  Object.keys(objArray).forEach((key) => (objArray[key] == null) && delete objArray[key]);
+                  const seen = new Set();
+                  objArray = objArray.filter(el => {
+                    const duplicate = seen.has(el.id);
+                    seen.add(el.id);
+                    return !duplicate;
+                  });
+                  objArray = objArray.filter(val => !!val.value)
+               }
+             } 
+           }
+           
+       }
+       return objArray;
+     }
+     return [];
+   }
+ 
+   export function getVarTypeAssoc(data: GraphNodes): { name: string; value: number }[] {
+     // Variant Gene Associations
+     if (!data) {
+       return [];
+     }
+     const result: { name: string; value: number }[] = [];
+     for (const neighbour of data.nodes[0].neighbors) {
+       if (neighbour.property !== "geneID") {
+         continue;
+       }
+       for (const node of neighbour.nodes) {
+         let variant = null;
+           for (const n of node.neighbors) {
+             if (n.property === "geneID") {
+               for(const n1 of n.nodes) {        
+                 for(const n2 of n1.neighbors) {
+                   if(n2.property === "geneticVariantFunctionalCategory") {
+                    for (const n3 of n2.nodes) {
+                    }
+                     let count = 0;            
+                     variant = n2.nodes[0].value;
+                     count = 1;
+                     result.push({ name: variant, value: count });
+                   } 
+                 }
+               }
+             } 
+           }
+           
+       }
+       var dataNew = [];
+       result.forEach(function (element) {
```

Comment:
I think a cleaner/easier way to do all of this would be to create a map outside the for loop where the key is variant name and value is the count. As you iterate through the data, you can access the count for the current variant and add 1 to it. Then once the for loop completes, you go through the map and create the array of data points.

Same thing for the next 2 functions in this file as those functions seem to follow the same pattern